### PR TITLE
Deprecate matplotlib.cm.LUTSIZE

### PR DESCRIPTION
**Pull request by timhoffm**
we can directly use `mpl.rcParams['image.lut']`.

### CODE REVIEW

File: lib/matplotlib/cm.py [hunk 1]
```diff
@@ -26,7 +26,16 @@
 from matplotlib._cm_listed import cmaps as cmaps_listed
```
Comment (tacaswell) [inline lib/matplotlib/cm.py:26]:
```suggestion
from matplotlib._cm_listed import cmaps as cmaps_listed
import functools
```

**Review by tacaswell — APPROVED** [review]

File: lib/matplotlib/cm.py [hunk 2]
```diff
@@ -23,10 +23,20 @@
 import matplotlib as mpl
 from matplotlib import _api, colors, cbook
 from matplotlib._cm import datad
+import functools
```
Comment (QuLogic) [inline lib/matplotlib/cm.py:26]: This belongs with `collections.abc` in the standard library import block.

Comment (timhoffm) [reply lib/matplotlib/cm.py:26]: Anybody is welcome to fix this and force-push. I won't be able to do it myself for a couple more days.

Comment (tacaswell) [reply lib/matplotlib/cm.py:26]: So playing with the new codespaces, review comments on branches that have PRs are inline which is pretty nice!

Comment (tacaswell) [reply lib/matplotlib/cm.py:26]: but the "mpl contributors can push to my PR branch" ACL does not seem to be forwarded yet.  Oh well.

**Comment by tacaswell** [comment]
and rebased.

**Comment by tacaswell** [comment]
Third times the charm, hopefully all of our CI systems have auto-kill setup.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**